### DISCUSSION TITLE
netrw from hell

### DISCUSSION THREAD

**Discussion by armyers** [LazyVim/LazyVim#2075] [Q&A] [answered]
Hi,

I have a habit of running `nvim *` in a directory. If that directory contains subdirectories, those will be opened in buffers but they will be netrw buffers, not neo-tree. The problem is, it's a major annoyance to deal with those netrw buffers. Once you switch to one, you can't get out of it unless you split/vsplit to open a new window, then use `:e one-of-the-regular-file-buffers`, then you can switch back to the netrw buffer and do `:bw`.

For example, I created this tree:

```
.
├── 0.sh
├── a
│   ├── x.sh
│   ├── y.tf
│   └── z.py
├── b
│   ├── m.py
│   └── n.sh
├── c
│   ├── r.py
│   └── s.sh
├── d
│   ├── t.tf
│   ├── u.sh
│   └── v.py
├── e.tf
├── f.tf
└── g.sh
```

when I run `nvim *` at the top level, the subdirs `a, b, c, and d` are opened in netrw buffers. But iIf I just run `nvim .` it's all fine because `.` is opened in a neo-tree buffer and that works great. Or, if I do `nvim a` it works fine as well. It's only when I run `nvim *` or try to open any combination of a subdir and a regular file (eg, `nvim a e.tf`), or more than one subdir as in `nvim a b`, that I get the netrw buffers.  How do I make it so it never uses netrw to open directories?

On a related note, when I do `nvim .` or `nvim a` or similar, I get a neo-tree buffer and an empty `No Name` buffer. I don't really need the empty `No Name` buffer, typically. Is there a way to just open a neo-tree buffer?

And, let me know if this is a case of:
```
  me: doctor, it hurts when I do this!
  doctor: well don't do that!
```

**Comment by jyuan0** [comment, marked as answer]
> How do I make it so it never uses netrw to open directories?

You can disable netrw by uncommenting `"netrwPlugin"` in the `disabled_pllugins` in `config/lazy.lua`. With it disabled, I get empty buffers rather than netrw bufffers when trying to open multiple subdirs, or a file and subdir. Opening a single subdir results in a neo-tree buffer.

> On a related note, when I do nvim . or nvim a or similar, I get a neo-tree buffer and an empty No Name buffer. I don't really need the empty No Name buffer, typically. Is there a way to just open a neo-tree buffer?

With netrw disabled, this doesn't happen for me.

**Reply by armyers** [reply]
yes, that is much better! 👍 

if I run `nvim *` it opens all of the regular files, of course, but the subdirs are just empty buffers. When I switch `]b` into one of the subdir buffers, it's just empty and I can switch to the next buffer with `]b`. But if I open a neo-tree buffer/window `<leader>e` and switch to a subdir buffer, it opens that subdir in the neo-tree buffer, and the subdir buffer is closed/deleted/wiped. Then I switch to the next subdir buffer and same thing.

That works. Thank you!
